content/en/docs/components/pipelines/sdk/v2/v2-component-io.md

```diff
@@ -0,0 +1,129 @@
++++
+title = "Kubeflow Pipelines v2 Component I/O"
+description = "Differences between artifacts and parameters, and how to migrate an existing pipeline to be v2 compatible."
+weight = 20
++++
+
+{{% beta-status
+  feedbacklink="https://github.com/kubeflow/pipelines/issues" %}}
+
+## Artifacts and Parameters
+
+The Kubeflow Pipelines SDK v2 makes a distinction between inputs and outputs that are _parameters_ and those that are _artifacts_.
+
+* Parameters represent values.
+  They are inputs or outputs of type `str`, `int`, `float`, `bool`, `dict`, or `list` that typically are used to change the behavior of a pipeline. 
+  Parameters are stored in ML Metadata, thus you can query ML Metadata for the values of parameters.
+  Input parameters are always passed by value, which means that they are inserted into the command used to execute the component. 
+
+* Artifacts represent references to objects--most commonly files--that are produced during pipeline executions.
+  Examples of artifacts include `Model`, `Datasets`, `Metrics`, and so on.
+  Regardless of the types of artifacts, all artifacts have a `name`, a `uri`, and optionally some `metadata`.
+  Most commonly, the `uri` of an artifact is a cloud storage URI such as a Google Cloud Storage URI, or an Amazon Web Services S3 URI. 
+  The `name`, `uri`, and `metadata` of artifacts are stored in ML Metadata, but not the contents referenced by the `uri`.
+  Input artifacts are always passed by reference, i.e. the `uri` of an artifact or a JSON object containing the `name`, `uri`, and `metadata` of an artifact.
+
+Whether an input/output is a parameter or an artifact is decided by its type annotation. Inputs/outputs typed as `str`, `int`, `float`, `bool`, `dict`, or `list` are parameters, while everything else are artifacts. Kubeflow Pipelines SDK v2 defines a list of system artifact types including `Model`, `Dataset`, `Metrics`, `ClassificationMetrics`, `SlicedClassificationMetrics`, `HTML`, `Markdown`, and the generic `Artifact`. All unknown types including non-type are treated the same as the generic `Artifact` type.
+
+Artifacts and parameters cannot be passed to each other. By default, when compiling a pipeline with option `type_check=True`, only inputs and outputs with the exact same type can be connected together with the exception that the generic `Artifact` type is compatible with any artifact types.
+
+## Migrating from v1 components
+
+### Review and update inputs/outputs types.
```

Comment:
I feel it would be easier for users to visualize the difference directly if we have an example code snippet of what's working in v1 will not be working in v2 here for input/output types?